utils/ccdatofhir/mapping_resource_diagnostice_report.bal

```diff
@@ -0,0 +1,93 @@
+# + return - FHIR Diagnostic Report
+public isolated function mapCcdaDiagnosticReportToFhir(xml organizerElement) returns r4:DiagnosticReport? {
+    if isXMLElementNotNull(organizerElement) {
+        r4:DiagnosticReport diagnosticReport = {code: {}, status: "partial"};
+
+        xml idElement = organizerElement/<v3:id|id>;
+        xml statusCodeElement = organizerElement/<v3:statusCode|statusCode>;
+        xml codeElement = organizerElement/<v3:code|code>;
+        xml translationElement = codeElement/<v3:translation|translation>;
+        xml effectiveTimeLowElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:low|low>;
+        xml effectiveTimeHighElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:high|high>;
+        xml effectiveTimeValueElement = organizerElement/<v3:effectiveTime|effectiveTime>/<v3:value|value>;
+
+        int index = 0;
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if (mapCcdaIdToFhirIdentifierResult is r4:Identifier) {
+                diagnosticReport.identifier[index] = mapCcdaIdToFhirIdentifierResult;
+                index = index + 1;
+            }
+        }
+
+        diagnosticReport.status = mapCcdatoFhirDiagnosticReportStatus(statusCodeElement);
+
+        r4:CodeableConcept? mapCcdaCodingtoFhirCodeableConceptResult = mapCcdaCodingtoFhirCodeableConcept(translationElement);
+        if mapCcdaCodingtoFhirCodeableConceptResult is r4:CodeableConcept {
+            diagnosticReport.code = mapCcdaCodingtoFhirCodeableConceptResult;
+        } else {
+            mapCcdaCodingtoFhirCodeableConceptResult = mapCcdaCodingtoFhirCodeableConcept(codeElement);
+            if mapCcdaCodingtoFhirCodeableConceptResult is r4:CodeableConcept {
+                diagnosticReport.code = mapCcdaCodingtoFhirCodeableConceptResult;
+            }
+        }
+
+        r4:dateTime? mapCCDAEffectiveLowTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeLowElement);
+        if mapCCDAEffectiveLowTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectivePeriod.'start = mapCCDAEffectiveLowTimetoFHIRDateTimeResult;
+        }
+
+        r4:dateTime? mapCCDAEffectiveHighTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeHighElement);
+        if mapCCDAEffectiveHighTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectivePeriod.end = mapCCDAEffectiveHighTimetoFHIRDateTimeResult;
+        }
+
+        r4:dateTime? mapCCDAEffectiveValueTimetoFHIRDateTimeResult = mapCcdaDateTimeToFhirDateTime(effectiveTimeValueElement);
+        if mapCCDAEffectiveValueTimetoFHIRDateTimeResult is r4:dateTime {
+            diagnosticReport.effectiveDateTime = mapCCDAEffectiveValueTimetoFHIRDateTimeResult;
+        }
+        return diagnosticReport;
+    } else {
+        return ();
+    }
```

Comment:
```suggestion
    }
```

Reply:
This change would introduce `this function should explicitly return a value` warning to function signature. Also, `()` is explicitly returned so that, we can check for the type before appending to response and avoid if null. This would help to avoid appearing unnecessary empty fields in the final payload.

Reply:
+1
I think we can still get rid of the `else` here since simply `return ()`

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/a482466dc4ba847b3e57e8e27d7d0e8cc48cf07c